src/Tickets/Commerce/Gateways/Free/Gateway.php

```diff
@@ -64,14 +64,23 @@ public static function is_active(): bool {
 	}
 
 	/**
-	 * @inheritDoc
+	 * Determine whether the gateway should be shown as an available gateway.
+	 *
+	 * @since 5.1.6
+	 * @since TBD Prevented the cart totals from being recalculated multiple times.
+	 *
+	 * @return bool Whether the gateway should be shown as an available gateway.
 	 */
 	public static function should_show(): bool {
+		static $cart_total = null;
```

Comment:
Same comment about caching. Prefer tribe_cache()

Reply:
Same concern as the above cache comment. We had this [cached](https://github.com/the-events-calendar/event-tickets/blob/6ef2fe23d8b80f86707b410ee97e194278061fb7/src/Tickets/Commerce/Cart/Abstract_Cart.php#L81).